src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -2466,5 +2466,19 @@ ITypeSymbol ITypeSymbolInternal.GetITypeSymbol()
         internal abstract bool IsRecord { get; }
 
         internal abstract bool IsRecordStruct { get; }
+
+#nullable enable
+        internal abstract bool IsExtension { get; }
+
+        internal TypeSymbol? ExtensionUnderlyingTypeNoUseSiteDiagnostics
+            => IsExtension ? ExtensionUnderlyingTypeNoUseSiteDiagnosticsCore : null;
```

Comment:
>ExtensionUnderlyingTypeNoUseSiteDiagnosticsCore [](http://example.com/codeflow?start=29&length=47)

I am not sure what is the advantage of introducing this indirection instead of making ```ExtensionUnderlyingTypeNoUseSiteDiagnostics``` abstract.

Reply:
The advantage was to centrally enforce the invariant: only extension types can has an extension underlying type.